File: examples/sdk/python/verify_proofpack.py
from __future__ import annotations

import json
import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parents[3]
sys.path[:0] = [str(ROOT), str(ROOT / "backend"), str(ROOT / "packages/parva-python")]

from app.membranes.proofpack import verify_proof_pack  # noqa: E402


def main(path: str = "examples/external/proofpacks/civil-conversion.proofpack.json") -> int:
    artifact = json.loads((ROOT / path).read_text(encoding="utf-8"))
    ok, reason = verify_proof_pack(artifact)
    boundary = artifact.get("boundary") or (artifact.get("membrane") or {}).get("boundary", {})
    print(json.dumps({"verified": ok, "reason": reason, "boundary": boundary}, indent=2, sort_keys=True))
    return 0 if ok else 1


if __name__ == "__main__":
    raise SystemExit(main(sys.argv[1] if len(sys.argv) > 1 else "examples/external/proofpacks/civil-conversion.proofpack.json"))

File: examples/sdk/js/verifyProofPack.ts
import { readFileSync } from 'node:fs';
import { resolve } from 'node:path';
import { verifyProofPack } from '../../../packages/parva-local-kernel/src/index';

const path = process.argv[2] ?? 'examples/external/proofpacks/civil-conversion.proofpack.json';
const artifact = JSON.parse(readFileSync(resolve(path), 'utf8'));
const result = await verifyProofPack(artifact, []);

console.log(JSON.stringify({ ...result, boundary: artifact.boundary ?? artifact.membrane?.boundary }, null, 2));
process.exit(result.verified ? 0 : 1);
